test/mock-server-tester.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  CallType,
+  GrpcErrorType,
+  SuccessType,
+} from '../mock-server/datastore-server';
+
+const grpc = require('@grpc/grpc-js');
+import {ServiceError} from 'google-gax';
+import {Server} from '@grpc/grpc-js';
+
+/**
+ * Simulates a non-retryable error response from a gRPC service.
+ *
+ * This function is designed for use in mock server setups during testing.
+ * When invoked as a gRPC endpoint handler, it responds with a pre-configured
+ * error object representing a non-retryable error. The error includes a
+ * NOT_FOUND code (5), a fixed details message, and some metadata.
+ *
+ * @param {any} call - The gRPC call object, representing the incoming request.
+ * @param {function} callback - The callback function to be invoked with the
+ *   simulated error response. It expects two arguments: an error object and an
+ *   empty object (representing no response data).
+ */
+export function sendNonRetryableError(
+  call: CallType,
+  callback: (arg1: GrpcErrorType, arg2: SuccessType) => {},
+) {
+  const metadata = new grpc.Metadata();
+  metadata.set(
+    'grpc-server-stats-bin',
+    Buffer.from([0, 0, 116, 73, 159, 3, 0, 0, 0, 0]),
+  );
+  const error = new Error('error message') as ServiceError;
+  error.code = 5;
+  error.details = 'error details';
+  error.metadata = metadata;
+  callback(error, {});
+}
+
+export class ErrorGenerator {
+  private errorSeriesCount = 0;
+
+  /**
+   * Generates an error object for testing purposes.
+   *
+   * This method creates a `ServiceError` object, simulating an error
+   * that might be returned by a gRPC service. The error includes a
+   * `DEADLINE_EXCEEDED` code (4), a details message indicating the number of
+   * errors generated so far by this instance, and some metadata.
+   *
+   * @returns {ServiceError} A `ServiceError` object representing a simulated
+   *   gRPC error.
+   */
+  generateError() {
```

Comment:
nit - without reading the comment, it's actually hard interpret `generateError` to generate a `DEADLINE_EXCEEDED` error.

Reply:
I restructured this so that we pass the error code into generateError. This might make the intent of this method more clear.